provider/consul/consul_catalog.go

```diff
@@ -28,11 +29,14 @@ var _ provider.Provider = (*CatalogProvider)(nil)
 
 // CatalogProvider holds configurations of the Consul catalog provider.
 type CatalogProvider struct {
-	provider.BaseProvider `mapstructure:",squash"`
-	Endpoint              string `description:"Consul server endpoint"`
-	Domain                string `description:"Default domain used"`
-	Prefix                string `description:"Prefix used for Consul catalog tags"`
-	client                *api.Client
+	provider.BaseProvider      `mapstructure:",squash"`
+	Endpoint                   string `description:"Consul server endpoint"`
+	Domain                     string `description:"Default domain used"`
+	Prefix                     string `description:"Prefix used for Consul catalog tags"`
+	FrontEndRule               string `description:"Frontend rule used for Consul services"`
+	AllTagsConstraintFiltering bool   `description:"Should use all tags for constraint filtering"`
+	client                     *api.Client
+	FrontEndRuleTemplate       *template.Template
```

Comment:
@ldez The `FrontEndRuleTemplate` is the actual `Template` object that is not provided via configuration, but initialized from `FrontEndRule` value.  So I see it being similar to `api.Client` kind.  Guess it should probably be made lower-cased name `frontEndRuleTemplate` instead to clearly indicate that, correct?